### ISSUE TITLE
GAN implementation details

### ISSUE THREAD

**Issue by murrman95** [yxgeee/SDA#3]
Hello,

I'm working on implementing this paper. I've taken the code from your MMT-plus repository, and refitted it such that the training algorithm fits Alg(1) from the SDA paper (translating batched on the fly during joint training). Also, I'm using a single mean teacher for the encoder updates with kmeans instead of MMT (just trying to recreate results the original SDA paper results). In the paper the details for the implementation are pretty clear, but the cycle GAN architecture isn't really mentioned. Could you give details as to which GAN style you're using (LSGAN, Wasserstein, spectral, etc), which architectures you're using for generators, discriminators, if the learning rate decay is applied to both the cycleGAN and the target encoder. If you want, we could call and discuss. Let me know.
